core/src/main/java/com/linecorp/armeria/server/DefaultServerErrorHandler.java

```diff
@@ -86,8 +88,22 @@ public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable caus
         }
 
         if (cause instanceof RequestTimeoutException) {
-            return internalRenderStatus(ctx, ctx.request().headers(),
-                                        HttpStatus.SERVICE_UNAVAILABLE, cause);
+            final HttpStatus status;
+            if (ctx.method() == HttpMethod.GET) {
+                // Consider GET requests as having received the request fully.
+                status = HttpStatus.SERVICE_UNAVAILABLE;
```

Comment:
My understanding is that this branch is used for performance optimization. Because, generally, `request.isNormallyClosed()` should be true for `GET` method.

Reply:
A GET request is allowed to have a request body, thus `isNormallyClosed` might not be called.
https://stackoverflow.com/questions/978061/http-get-with-request-body

Reply:
Is there any reason that we should return 503 even if a request hasn’t fully received for a GET instead of 408?

Reply:
I thought we could consider a GET request as having received the request fully.
Do you prefer to return 408 instead?

Reply:
I thought returning 408 is more consistent behavior because that does not rely on HTTP method but check if a request is fully received.

Reply:
Alright then let me return 408 instead. For the record, a GET request that doesn't have any request body might get 408 as well because this `DefaultServerErrorHandler` can be called before `isNormalClosed()` is called.

Reply:
I think, in most cases, `isNormalClosed()` is true when a `DecodedHttpRequest` is created for a GET request. `endOfStream` would be true with a GET request having an empty body. As a result, `EmptyContentDecodedHttpRequest` will be created for which `isNormalClosed()` is always true.

Reply:
That is true. I thought an HTTP/1.1 request was handled differently but it was not after all.